adamsjr8576/Refactor-Tractor
# Inheritance

@Jessmaxim303 @Dustin-Mikusko @khalidwilliams @brittanystoroz 

**What does this PR do??**

 - This PR creates a parent class of userFitness that is then inherited by the Sleep, Hydration and Activity classes. This is meant to DRY up the code base and take all repeating functions into the parent class. It also made it possible to refactor some of the existing methods on the child classes in regards to how they receive the user data pertinent to that class.

- This PR updates the tests accordingly including making a new test file for the userFitness class.

- This PR updates the script.js file according to the changes made on each class and does not affect the ux of the site at all.

**Where should the reviewer start?**

 - The viewer should start by pulling down and looking at the userFitness.js file, which is the parent of the Sleep, Hydration and Activity classes. Then look at how that affects those child classes, specifically how they interact with one another. It is helpful to use the debugger to see how JS is stepping through these files (instantiating the child classes), which helps to show how it is working and connected.

**How should this be manually tested?**

- Pull down this branch, run both npm run test and npm run start to make sure your local machine passes the tests, and the localhost shows the script.js updates.

**What are the relevant tickets?** #25
